src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (!(target instanceof Carnivore) && target.hidden !== true) {
```

Comment:
why you check whether the animal is not carnivore? To me it makes sense to check whether it is the Herbivore

Reply:
also, why `!== true`? Why not `===` false?

Anyway, let's use shortcut for this boolean, like `if (boolean)` of `if (!boolean)`